Resources/public/js/components/products/components/attributes/main.js

```diff
@@ -0,0 +1,365 @@
+        },
+
+        /**
+         * Create the overlay
+         */
+        createAddOverlay = function() {
+
+            // call JSON to get the attributes from the server then create the overlay after it's done
+            var jqxhr = $.getJSON("api/attributes", function(data) {
+                this.attrTypes = new Array();
+
+                $.each(data._embedded.attributes, function(key, value) {
+                    var newAttribute = {
+                        "id": value.id,
+                        "name": value.name
+                    };
+
+                    // at this time we support only text type attributes
+                    if (value.type.name === "product.attribute.type.text") {
+                        this.attrTypes.push(newAttribute);
+                    }
+
+                }.bind(this));
+
+            }.bind(this))
+            .done(function() {
+                // create container for overlay
+                var $overlay = this.sandbox.dom.createElement('<div>');
+                this.sandbox.dom.append(this.$el, $overlay);
+
+                // create content
+                this.sandbox.start([
+                    {
+                        name: 'overlay@husky',
+                        options: {
+                            el: $overlay,
+                            supportKeyInput: false,
+                            title: this.sandbox.translate('product.attribute.overlay.title'),
+                            skin: 'normal',
+                            openOnStart: true,
+                            removeOnClose: true,
+                            instanceName: overlayInstanceName,
+                            data: createAddOverlayContent.call(this),
+                            okCallback: overlayOkClicked.bind(this)
+                        }
+                    }
+                ]);
+
+            }.bind(this))
+            .fail(function() {
+                console.log("Error retrieving attributes from server");
+            });
+
+            jqxhr.complete(function() {
+                // create dropbox in overlay
+                var selectOptions = {
+                    el: '#selectBox',
+                    instanceName: selectInstanceName,
+                    multipleSelect: false,
+                    defaultLabel: this.sandbox.translate('product.attribute.overlay.defaultlabel'),
+                    valueName: 'name',
+                    isNative: true,
+                    data: this.attrTypes
+                };
+
+                this.sandbox.start([
+                    // start select
+                    {
+                        name: 'select@husky',
+                        options: selectOptions
+                    }
+                ]);
+
+                // define select event for dropbox
+                this.sandbox.on('husky.select.' + selectInstanceName + '.selected.item', function(item) {
+                    attrId = parseInt(item);
```

Comment:
i'd prefer using attrId 
* not shortened and
* as this variable (this.attributeId)
  * if you do this, you'd need to bind this context to callback functions (like this one)